lib/availability.ts

```diff
@@ -45,3 +52,61 @@ export function getAvailabilityFromSchedule(schedule: Schedule): Availability[]
     return availability;
   }, [] as Availability[]);
 }
+
+/**
+ * PostgreSQL stores the date relatively
+ */
+export function getWorkingHours(
+  timeZone: string,
+  availability: Pick<Availability, "days" | "startTime" | "endTime">[]
+) {
+  // clearly bail when availability is not set, set everything available.
+  if (!availability.length) {
+    return [
+      {
+        days: [0, 1, 2, 3, 4, 5, 6],
+        // shorthand for: dayjs().startOf("day").tz(timeZone).diff(dayjs.utc().startOf("day"), "minutes")
+        startTime: 0,
+        endTime: 1439,
+      },
+    ];
+  }
+
+  const workingHours = availability.reduce((workingHours: WorkingHours[], schedule) => {
+    const startTime = dayjs
+      .utc(dayjs.utc(schedule.startTime).format("HH:mm:ss"), "HH:mm:ss")
+      .tz(timeZone, true)
+      .diff(dayjs.utc().startOf("day"), "minutes");
+    const endTime = dayjs
+      .utc(dayjs.utc(schedule.endTime).format("HH:mm:ss"), "HH:mm:ss")
+      .tz(timeZone, true)
+      .diff(dayjs.utc().startOf("day"), "minutes");
+    // add to working hours, keeping startTime and endTimes between bounds (0-1439)
+    workingHours.push({
+      days: schedule.days,
+      startTime: Math.max(0, Math.min(1439, startTime)),
```

Comment:
Remove magic numbers

**In constants somewhere, at top or file or in a `constants.tsx` file**
```ts
const MINUTES_PER_DAY = 60*24;
const ONE_MINUTE_BEFORE_MIDNIGHT = MINUTES_PER_DAY - 1

```suggestion
      startTime: Math.max(0, Math.min(ONE_MINUTE_BEFORE_MIDNIGHT, startTime)),
````

Reply:
Yeah this is a good idea, I'll add some comments as well because `ONE_MINUTE_BEFORE_MIDNIGHT` is actually what you get when you do - So I'll name this accordingly. (END_OF_DAY)
```
dayjs.utc().endOf("day").get("hour") * 60 + ...get("minute") // 59 = 1439.